src/logic_utils/truth_table.cpp

```diff
@@ -0,0 +1,177 @@
+    char c = expression[i];
+
+    if (isspace(c))
+      continue;
+
+    if (isdigit(c)) {
+      std::string num;
+      while (i < expression.length() && (isdigit(expression[i]) || expression[i] == '.')) {
+        num += expression[i];
+        ++i;
+      }
+      --i;
+      values.push(std::stod(num));
+    } else if (c == '(') {
+      operators.push(c);
+    } else if (c == ')') {
+      while (!operators.empty() && operators.top() != '(') {
+        double b = values.top();
+        values.pop();
+
+        double a = values.top();
+        values.pop();
+
+        char op = operators.top();
+        operators.pop();
+
+        values.push(applyOperation(a, b, op));
+      }
+
+      if (!operators.empty())
+        operators.pop(); // Discard the opening parenthesis
+    } else if (isOperator(c)) {
+      while (!operators.empty() && getPrecedence(c) <= getPrecedence(operators.top())) {
+        double b = values.top();
+        values.pop();
+
+        double a = values.top();
+        values.pop();
+
+        char op = operators.top();
+        operators.pop();
+
+        values.push(applyOperation(a, b, op));
+      }
+
+      operators.push(c);
+    }
+  }
+
+  while (!operators.empty()) {
+    double b = values.top();
+    values.pop();
+
+    double a = values.top();
+    values.pop();
+
+    char op = operators.top();
+    operators.pop();
+
+    values.push(applyOperation(a, b, op));
+  }
+  cout << values.top() << endl;
+  return values.top();
+}
+
+
+
+string replaceVariables(const string & expression, unordered_map < char, bool > variableValues) {
+  string result;
+  for (char token: expression) {
+    if (isalpha(token)) {
+      // If the token is a variable, append the corresponding value (0 or 1) to the result string
+      result += variableValues[token] ? '1' : '0';
+    } else {
+      // If the token is not a variable, append it as it is
+      result += token;
+    }
+  }
+  return result;
+}
+
+void expressionIterator(int num_var, string expression, unordered_map < char, bool > variableValues) {
+  if (num_var == 0) {
+    expression = replaceVariables(expression, variableValues);
+    evaluateExpression(expression);
+  } else {
+    variableValues[variables[num_var-1]] = 0;
+    expressionIterator(num_var - 1, expression, variableValues);
+    variableValues[variables[num_var-1]] = 1;
+    expressionIterator(num_var - 1, expression, variableValues);
+  }
+}
+
+unordered_map < char, bool > countUniqueVariables(const string & expression) {
+  unordered_map < char, bool > variables;
+
+  for (char token: expression) {
+    if (isalpha(token)) {
+      variables[token] = 0;
+    }
+  }
+  return variables;
+}
+
+map<int,char> uniqueVariablesIndex(const string & expression) {
+  map<int,char> variables;
+  int i = 0;
+  for (char token: expression) {
+    if (isalpha(token)) {
+      variables.insert(make_pair(i,token));
+      i++;
+    }
+  }
+  return variables;
+}
+
+void truth table(){
```

Comment:
I guess this function should return the list to the caller instead of printing it.